Support in progress response for async requests

emilyki opened the pull request:
This pull request addresses the issue #493.

The change addresses support for the `HTTP 202` response (which indicates that a request was accepted for processing, but was not completed) for Cruise Control asynchronous requests.

This is an example log output from testing the 202 response:
![test](https://user-images.githubusercontent.com/15850091/62180272-703d4d80-b31d-11e9-9e9b-50902493231f.png)

emilyki commented:
It seems like the build has failed because the unit tests aren't complete yet. I'll address this issue once the proposed implementation/approach is approved. Thanks!

emilyki commented:
May I ask for your feedback/comments on the approach proposed @efeg @kidkun ? Thanks!

kidkun commented:
Thanks for the well-organized and detailed design doc at issue #493 !
I think returning http-202 is a good idea to indicate the calculation is not done and inform client to poll later.

Note recently we have a refactor of servlet code and make in-progress response a top-level response(create a Class [ProgressResult](https://github.com/linkedin/cruise-control/blob/master/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/response/ProgressResult.java)), I think instead of adding a new `writeInProgressResponse()` in interface(`CruiseControlResponse`) we can directly override method `writeSuccessResponse()` in `ProgressResult` to return `SC_ACCEPTED` in header.

emilyki commented:
Thanks for your suggestions @kidkun! I'll update the pull request with the new approach (as soon as my final exams are done, by the end of this week).

emilyki commented:
Hello @kidkun and @efeg, I've rebased the branch to the latest master and updated the implementation to override the `writeSuccessResponse` method of `ProgressResult` instead. May I ask for another review on this? Thanks!
